### PR TITLE
Add powershell completion

### PR DESCRIPTION
**Pull request by douglaswth**

### CODE REVIEW

File: pkg/cmd/kind/completion/powershell/powershell.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (stmcginnis) [inline pkg/cmd/kind/completion/powershell/powershell.go:2]: Since this is newly added code, the copyright year should probably be 2024.

**Review by stmcginnis — COMMENTED** [review]
Seems reasonable since we do publish Windows binaries.

/lgtm
/ok-to-test

**Comment by stmcginnis** [comment]
/lgtm

Thanks!

File: pkg/cmd/kind/completion/completion.go [hunk 2]
```diff
@@ -69,6 +71,9 @@ This depends on the bash-completion binary.  Example installation instructions:
 # for fish users
 	% kind completion fish > ~/.config/fish/completions/kind.fish
 
+# for powershell users
+	PS> kind completion powershell | Out-String | Invoke-Expression
```
Comment (BenTheElder) [inline pkg/cmd/kind/completion/completion.go:75]: for reference: https://github.com/spf13/cobra/blob/main/site/content/completions/_index.md#creating-your-own-completion-command

**Comment by BenTheElder** [comment]
/lgtm
/approve
thanks!

**Pull request merged**